trulens_eval/trulens_eval/feedback.py

```diff
@@ -1,5 +1,353 @@
 """
 # Feedback Functions
+
+The `Feedback` class contains the starting point for feedback function
+specification and evaluation. A typical use-case looks like this:
+
+```python 
+from trulens_eval import feedback, Select, Feedback
+
+openai = feedback.OpenAI()
+
+f_qs_relevance = Feedback(openai.qs_relevance)
+    .on_input()
+    .on(Select.Record.app.combine_docs_chain._call.args.inputs.input_documents[:].page_content)
+    .aggregate(numpy.min)
+```
```

Comment:
**Rick Comment**: lets start with the simplest case at the top; and add more details if needed.

**Possible Edit below:**

```
f_qs_relevance = Feedback(openai.language_match)
    .on_input(), on(Select.Output) #Todo fix syntax
```
The components of this specifications are:
- **Provider classes** -- `feedback.OpenAI` contains feedback function implementations
  like `language_match`. Other classes subtyping `feedback.Provider` include
  `Huggingface` and `Cohere`.
- **Feedback implementations** -- `openai.language_match` is a feedback function
  implementation. Here it is a method with the following signature:
    ```python
    def language_match(self, llm_input: str, llm_output: str) -> float:
    ```
    That is, `language_match` is a plain python method that accepts a `llm_input`
    and a `llm_output`, both strings, and produces a float (assumed to be between
    0.0 and 1.0).
- **Feedback constructor** -- The line `Feedback(openai.language_match)` constructs a
  Feedback object with a feedback implementation.
- **Argument specification** -- The next two lines, `on_input` and `on`, specify how
  the `language_match` arguments are to be determined from an app record or app
  definition. The general form of this specification is done using `on` but
  several shorthands are provided. `on_input` states that the first argument to
  `language_match` (`llm_input`) is to be the main input of the app. 

## Fine-grained Selection and Aggregation

For more advanced control on the feedback function operation, we allow data selection and aggregation.

```
f_qs_relevance = Feedback(openai.qs_relevance)
    .on_input()
    .on(Select.Record.app.combine_docs_chain._call.args.inputs.input_documents[:].page_content)
    .aggregate(numpy.min)
```

**Argument Selection specification** -- Where we previously set, `on_input` and `on`, the
  `on(Select...)` line enables specification of where the `statement` argument to the
  implementation comes from. The form of the specification will be discussed in
  further details in the **Specifying Arguments** section.